fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java

```diff
@@ -536,4 +569,273 @@ private static <T extends ScalarOperator> void collect(ScalarOperator root, Clas
 
         root.getChildren().forEach(child -> collect(child, clazz, output));
     }
+
+    public static void getRelatedMvs(List<Table> tablesToCheck, Set<MaterializedView> mvs) {
+        Set<MvId> newMvIds = Sets.newHashSet();
```

Comment:
Maybe add a limit/threshold to avoid recursive too deeply?

Reply:
ok